File: src/main/java/hu/frontrider/arcana/content/items/formula/text/FormulaText.java
package hu.frontrider.arcana.content.items.formula.text;

import net.minecraft.client.util.ITooltipFlag;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.world.World;

import javax.annotation.Nullable;
import java.util.List;

/**
 * @author Kis András Gábor
 * 2018.08.28.
 */
@FunctionalInterface
public interface FormulaText {
    
    void addText(ItemStack stack, NBTTagCompound tagCompound, @Nullable World worldIn, List<String> tooltip);
}

File: src/main/java/hu/frontrider/arcana/sided/client/gui/formula/text/FormulaText.kt
package hu.frontrider.arcana.sided.client.gui.formula.text

import net.minecraft.client.util.ITooltipFlag
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.world.World

/**
 * @author Kis András Gábor
 * 2018.08.28.
 */
@FunctionalInterface
interface FormulaText {

    fun addText(stack: ItemStack, tagCompound: NBTTagCompound, worldIn: World?, tooltip: MutableList<String>)
}
